charon/storage.py

```diff
@@ -228,13 +241,30 @@ def path_upload_handler(full_file_path: str, path: str):
             try:
                 if not self.dry_run:
                     if need_overwritten:
-                        fileObject.put(
+                        file_object.put(
                             Body=open(full_file_path, "rb"),
                             Metadata=f_meta,
                             ContentType=content_type
                         )
+
+                        if digests:
+                            bucket.Object(full_file_path + ".sha1").put(
```

Comment:
I think these are only needed for maven pkg(maven-metadata.xml, archetype-catalog.xml), right? So maybe a better way is extracted these logic to charon.pkgs.maven module, but not in the storage module. I'll check how to change this today.